src/parser/lex.rs

```diff
@@ -0,0 +1,175 @@
+// L1 Compiler
+//! Lexer
+// Author: Miles Conn <mconn@andrew.cmu.edu>
+
+// Update this file to lex the necessary keywords and other tokens
+// in order to make the grammar forward compatible with C0.
+// Note this project relies on logos 0.12.1 see docs [here]
+// (https://docs.rs/logos/0.12.1/logos/index.html)
+
+#![allow(clippy::upper_case_acronyms)]
+use enum_as_inner::EnumAsInner;
+use logos::{Lexer, Logos, Skip};
+use std::{fmt, num::ParseIntError};
+use strum_macros::AsRefStr;
+
+
+fn from_num<'b>(lex: &mut Lexer<'b, Token<'b>>) -> Result<i32, String> {
+    let slice = lex.slice();
```

Comment:
We are using this for just the comments right? maybe we just only move the relevant parts.